src/jit/morph.cpp

```diff
@@ -16851,6 +16857,8 @@ void Compiler::fgMorph()
     /* For x64 and ARM64 we need to mark irregular parameters early so that they don't get promoted */
     fgMarkImplicitByRefArgs();
 
+    EndPhase(PHASE_MORPH4);
+
     /* Promote struct locals if necessary */
     fgPromoteStructs();
```

Comment:
It might be useful to split these - `PHASE_MORPH_PROMOTE` and `PHASE_ADDR_EXP_LCLS` or something. But again, my intuition is probably not terribly reliable about how significant each of these may or may note be.